lib/index.js

```diff
@@ -123,11 +123,23 @@ exports.compose = function (manifest /*, [options], callback */) {
 
         var plugins = [];
 
-        Object.keys(manifest.plugins || {}).forEach(function (name) {
+        if (Array.isArray(manifest.plugins)) {
+            for (var i = 0, l = manifest.plugins.length; i < l; i++) {
+                var pluginObject = manifest.plugins[i];
+                var keys = Object.keys(pluginObject);
 
-            var plugin = internals.parsePlugin(name, manifest.plugins[name], options.relativeTo);
-            plugins = plugins.concat(plugin);
-        });
+                Hoek.assert(keys.length === 1, 'Invalid plugin config');
+
+                var plugin = internals.parsePlugin(keys[0], pluginObject[keys[0]], options.relativeTo);
+                plugins = plugins.concat(plugin);
```

Comment:
The 'plugin' value may or may not be an array.  If it is an array, it should be concatenated, not pushed.  The use of concat allows for either.